src/Game.cs

```diff
@@ -212,6 +212,9 @@ public GameWindow Window
 		private long previousTicks = 0;
 		private int updateFrameLag;
 		private bool forceElapsedTimeToZero = false;
+		private TimeSpan[] previousSleepTimes = new TimeSpan[100];
```

Comment:
If you make the array a power of two size, like 64, you can get rid of the expensive mod operation below and switch to a cheap bit mask.